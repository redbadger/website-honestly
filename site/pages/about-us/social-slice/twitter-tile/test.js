import React from 'react';
import { render } from 'enzyme';

import TwitterTile from './index';

const specialTweet = {
  text: 'I am a tweet &amp; I contain special characters',
  url: 'https://twitter.com/redbadgerteam/status/1108421453903523841',
  retweetCount: 0,
  favouriteCount: 5,
  created: '2019-03-20',
};

describe('TwitterTile', () => {
  it('renders correctly with special characters', () => {
    expect(render(<TwitterTile tweet={specialTweet} index={1} />)).toMatchSnapshot();
  });
});
