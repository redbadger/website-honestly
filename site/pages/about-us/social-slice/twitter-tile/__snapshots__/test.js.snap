// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TwitterTile renders correctly with special characters 1`] = `
<a
  class="link"
  href="https://twitter.com/redbadgerteam/status/1108421453903523841"
  rel="noopener noreferrer"
  tabindex="-1"
  target="_blank"
>
  <div
    class="twitter mauve"
    tabindex="0"
  >
    <div
      class="handle"
    >
      <svg
        class="twitterIcon"
        viewBox="0 0 25 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <title>
          Twitter
        </title>
        <path
          d="M7.771 19.709c9.117 0 14.104-7.52 14.104-14.04 0-.214 0-.427-.015-.638a10.06 10.06 0 0 0 2.473-2.555c-.905.4-1.864.661-2.847.777a4.956 4.956 0 0 0 2.18-2.73 9.96 9.96 0 0 1-3.148 1.198A4.977 4.977 0 0 0 14.563.75a4.932 4.932 0 0 0-2.492 5.472A14.094 14.094 0 0 1 1.855 1.066 4.925 4.925 0 0 0 3.39 7.653a4.937 4.937 0 0 1-2.25-.618v.063a4.942 4.942 0 0 0 3.977 4.837 4.97 4.97 0 0 1-2.238.085 4.96 4.96 0 0 0 4.63 3.427 9.974 9.974 0 0 1-7.335 2.045 14.079 14.079 0 0 0 7.597 2.213"
          fill-rule="evenodd"
        />
      </svg>
      <span
        class="handleText"
      >
        @RedBadger
      </span>
    </div>
    <div
      class="tweet tweetWithNoImage"
    >
      I am a tweet & I contain special characters
    </div>
    <div
      class="meta"
    >
      <span>
        <svg
          class="icon retweets"
          viewBox="0 0 27 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M5.364 15.704a.999.999 0 0 1-.996-1V6.568L0 6.565 5.935.069l5.936 6.498H7.498v5.995h4.696l3.077 3.142H5.364zm21.303-6.565h-4.37V.999A.998.998 0 0 0 21.302 0h-9.906l3.075 3.141h4.696v5.996l-4.371.002 5.936 6.497 5.935-6.497z"
            fill="#212121"
            fill-rule="evenodd"
          />
        </svg>
        0
      </span>
      <span>
        <svg
          class="icon"
          viewBox="0 0 17 16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M16.611 4.613C16.37 1.94 14.476 0 12.108 0c-1.577 0-3.022.85-3.834 2.21C7.468.832 6.083 0 4.53 0 2.163 0 .27 1.94.028 4.613c-.02.118-.098.74.141 1.753.345 1.462 1.142 2.791 2.304 3.844l5.797 5.26 5.896-5.26c1.162-1.053 1.959-2.382 2.304-3.844.239-1.013.16-1.635.141-1.753z"
            fill="#212121"
            fill-rule="evenodd"
          />
        </svg>
        5
      </span>
    </div>
  </div>
</a>
`;
